### PR TITLE
Fix issue 36146

### PR DESCRIPTION
**Pull request by tonytonev**
Improved error message to clarify why a custom mapping is not being accepted. Fix for issue #36146.

### CODE REVIEW

**Review by tonytonev — COMMENTED** [review]
Fixed according to @mdanatg 's suggestion here: https://github.com/tensorflow/tensorflow/issues/36146#issuecomment-608541085

**Review by mdanatg — CHANGES_REQUESTED** [review]
Thanks for adding this! Just one suggestion to include as much context information as we can.

**Review by mdanatg — DISMISSED** [review]

**Review by tonytonev — COMMENTED** [review]
@mdanatg it looks like "Ubuntu Sanity" and "MacOS Python2 and CC" CI builds failed for this, but I can't if the reason. What should I do?

File: tensorflow/python/util/nest.py [hunk 1]
```diff
@@ -159,7 +159,13 @@ def _sequence_like(instance, args):
     tf_logging.log_first_n(
         tf_logging.WARN, "Mapping types may not work well with tf.nest. Prefer"
         "using MutableMapping for {}".format(instance_type), 1)
-    return instance_type((key, result[key]) for key in instance)
+    try:
+      return instance_type((key, result[key]) for key in instance)
+    except TypeError as err:
+      raise TypeError(f"Error creating an object of type {type(instance)} "
+                      f"like {instance}. Note that it must accept a single "
```
[the remark this answers is not in the document]
Comment (mdanatg) [reply tensorflow/python/util/nest.py:166]: I think the linter fomplains about the f-strings. Due to some internal restrictions, the code still need to be Python2-compatible for a few more months. So we need to use `.format()` instead, for now.

[the remark this answers is not in the document]
Comment (tonytonev) [reply tensorflow/python/util/nest.py]: OK, I updated it to use `.format()`

**Review by mdanatg — APPROVED** [review]

**Pull request merged**